package/carbonio-proxy

```diff
@@ -1,83 +1,68 @@
 #!/bin/bash
+SERVICE="proxy"
+SERVICE_COMMON_NAME="Carbonio ${SERVICE}"
+SERVICE_NAME="carbonio-${SERVICE}"
+SERVICE_USER="carbonio-${SERVICE}"
+SERVICE_GROUP="carbonio-${SERVICE}"
+SERVICE_BASE_DIR="/etc/carbonio/${SERVICE}/service-discover/"
 
 if [[ $(id -u) -ne 0 ]]; then
   echo "Please run as root"
   exit 1
 fi
 
 if [[ "$1" != "setup" ]]; then
-  echo "Syntax: carbonio-proxy <setup> to automatically setup the service"
+  echo "Syntax: '${SERVICE_NAME} setup' to automatically configure the service"
   exit 1;
 fi
 
-# decrypt the bootstrap token, asking the password to the sys admin
+# Decrypt the bootstrap token, asking the password to the sys admin
 # --setup check for SETUP_CONSUL_TOKEN env. variable and uses it
-# to avoid re-asking for the password multiple times
-echo -n "insert the cluster credential password: "
+# to avoid re-asking for the password
+echo -n "Insert the cluster credential password: "
+# shellcheck disable=SC2155
 export CONSUL_HTTP_TOKEN=$(service-discover bootstrap-token --setup)
 EXIT_CODE="$?"
 echo ""
 if [[ "${EXIT_CODE}" != "0" ]]; then
-  echo "cannot access to bootstrap token"
+  echo "Cannot access to bootstrap token"
   exit 1;
 fi
-# limit secret visibility as much as possible
+# Limit secret visibility as much as possible
 export -n SETUP_CONSUL_TOKEN
 
-POLICY_NAME='carbonio-proxy-policy'
-POLICY_DESCRIPTION='Carbonio Proxy service policy for config generation and for sidecar proxy'
-POLICY_RULES="$(cat <<EOF
-"key_prefix" = {
-  "carbonio-proxy/" = {
-    "policy" = "read"
-  }
-}
-"node_prefix" = {
-  "" = {
-    "policy" = "read"
-  }
-}
-"service" = {
-  "carbonio-proxy" = {
-    "policy" = "write"
-  }
-  "carbonio-proxy-sidecar-proxy" = {
-    "policy" = "write"
-  }
-}
-EOF
-)"
+POLICY_NAME="${SERVICE_NAME}-policy"
+POLICY_DESCRIPTION="Carbonio ${SERVICE} service policy for service and sidecar proxy"
 
-# create or update policy for the specific service (this will be shared across cluster)
-consul acl policy create -name "${POLICY_NAME}" -description "${POLICY_DESCRIPTION}" -rules "${POLICY_RULES}" >/dev/null 2>&1
+# Create or update policy for the specific service (this will be shared across cluster)
+consul acl policy create -name "${POLICY_NAME}" -description "${POLICY_DESCRIPTION}" -rules @"${SERVICE_BASE_DIR}/policies-${SERVICE}.json" >/dev/null 2>&1
+# shellcheck disable=SC2181
 if [[ "$?" != "0" ]]; then
-    consul acl policy update -no-merge -name "${POLICY_NAME}" -description "${POLICY_DESCRIPTION}" -rules "${POLICY_RULES}"
+    consul acl policy update -no-merge -name "${POLICY_NAME}" -description "${POLICY_DESCRIPTION}" -rules @"${SERVICE_BASE_DIR}/policies-${SERVICE}.json"
```

Comment:
Why name is policies-proxy? It should just be policies.json.
All files should be under /etc/carbonio/proxy/service-discover/

Reply:
Link to the repo? Thanks

Reply:
https://bitbucket.org/zextras/mongooseim-package